src/pkg/backup/details/details.go

```diff
@@ -403,7 +403,9 @@ const (
 	ExchangeEvent
 	ExchangeMail
 
-	SharePointItem ItemType = iota + 100
+	SharePointLibrary ItemType = iota + 100
+	SharePointList    ItemType = iota + 100
+	SharePointPage    ItemType = iota + 100
```

Comment:
Well, apparently I didn't fully grok iota when I set this up.  The current change will update both sharepoint (a benign breaking change) _and_ onedrive (an actual breaking change).  see: https://go.dev/play/p/0kmJBLzJ7ME

We should probably update this whole declaration set to the following (with some additional comments for future clarity):


```suggestion
// Item typing is enumerated by service (hundredth digit) and data type (ones digit).
// Ex: exchange is 00x where x is the data type.  Sharepoint is 10x, and etc.
// Every item info struct should get its own hundredth enumeration entry.
// Every item category for that service should get its own entry (even if differences
// between types aren't apparent on initial implementation, this future-proofs
// against breaking changes).  
const (
	UnknownType ItemType = iota // 0, global unknown value

	ExchangeContact
	ExchangeEvent
	ExchangeMail
)

const (
        SharePointLibrary ItemType = iota + 100
	SharePointList    ItemType
	SharePointPage    ItemType
)
	
// 205 is a result of prior bad use of iota.
// if we introduce future oneDrive types, they
// should increment from here.  eg: 205, 206...
const OneDriveItem ItemType = 205

// 306 is a result of prior bad use o iota.
// if we introduce future folder types, they
// should increment from here.  eg: 306, 307...
//
// The next additional service should start
// enumerating at iota+400
const FolderItem ItemType = 306
```

Reply:
as I posted elsewhere, I think we may be over-complicating this a bit. For things that already have numbers assigned, we need to keep those numbers (with the exception of sharepoint as this PR updates it). Otherwise, the assignment of numbers to services is a bit immaterial imo. The reason for enums is so that we don't have to remember random numbers

Sure, not having contiguous numbers for data types in a service makes things a bit less pretty to read if anyone happens to dump the raw JSON, but honestly I'm not sure it's even possible to print the JSON in Corso without filtering out some fields. You could print it using kopia, but at that point you're so far down the stack that having to do a little number translation is the least of your worries. But having a rule of "append new data types to the end of the enum set" and having a comment telling people not to rearrange the order of the enum set feels harder to accidentally misuse

Reply:
Totally fine with me.  Lets move it to concrete ints and drop the iota?

Reply:
having the iota is still fine as it makes appending new entries to the list dead simple. We just need to make sure we don't disturb prior values that need to be fixed. We could just
1. fixup SP numbers and leave their declarations where they are
2. for future additions, just add them to after the folder declaration

Reply:
Specifying an int const isn't dead simple?

Reply:
it can be, but I'm always afraid I'll accidentally type the wrong number and not notice (happens more than I'd like 😅 )

Reply:
Resovled with [0ccde14](https://github.com/alcionai/corso/pull/2720/commits/0ccde1491fe68459b3becb816cc3022d02e7c589)